packages/tiny-agents/src/lib/webServer.ts

```diff
@@ -0,0 +1,122 @@
+import type { IncomingMessage } from "node:http";
+import { createServer, ServerResponse } from "node:http";
+import type { AddressInfo } from "node:net";
+import { z } from "zod";
+import type { Agent } from "../index";
+import { ANSI } from "./utils";
+import { stdout } from "node:process";
+import type { ChatCompletionStreamOutput } from "@huggingface/tasks";
+
+const REQUEST_ID_HEADER = "X-Request-Id";
+
+const ChatCompletionInputSchema = z.object({
+	messages: z.array(
+		z.object({
+			role: z.enum(["user", "assistant"]),
```

Comment:
```suggestion
			role: z.enum(["system", "user", "assistant"]),
```

there can be `system` prompt, no?